File: Niveau 3/Niveau 3.10/decimal.py
#-------------------------------------------------------------------------------
# Name:        Lecture binaire
# Purpose:     France-IOI niveau 3.10
#
# Author:      Sébastien
#
# Created:     16/10/2022
#-------------------------------------------------------------------------------

def main():
    s = input()
    n = 0
    for c in s:
        n = 2*n + int(c)
    print(n)

if __name__ == '__main__':
    main()

File: Niveau 3/Niveau 3.10/decimal.hs
--------------------------------------------------------------------------------
-- Name:        Lecture binaire
-- Purpose:     France-IOI niveau 3.10
--
-- Author:      Sébastien
--
-- Created:     16/10/2022
--------------------------------------------------------------------------------

main :: IO()
main = do
    s <- getLine   
    print $ foldl (\n c -> 2*n + (if c == '1' then 1 else 0)) 0 s
